target/arm/translate-a64.c

```diff
@@ -14633,8 +15361,17 @@ static void disas_a64_insn(CPUARMState *env, DisasContext *s)
         }
     }
 
+    // Top 4 bits of op0
     switch (extract32(insn, 25, 4)) {
-    case 0x0: case 0x1: case 0x3: /* UNALLOCATED */
+    case 0x1: /* Morello Instructions */
```

Comment:
Not sure which cases you mean. Currently, 0x1 will fall through to unallocated_encoding, which is nice if target is not CHERI.

Reply:
Oh I see, I missed that in my pass through, could do with a `/* fallthrough */` comment. Though I'd be inclined to just rewrite it as:

```
case 0x1:
#ifdef TARGET_CHERI
    if (!disas_cheri(s, insn))
#endif
    {
        unallocated_encoding(s);
    }
    break;
```
and not need the fallthrough, then you can move this case to be next to 2 rather than first here.